torch/utils/cpp_extension.py

```diff
@@ -142,19 +142,49 @@ def _find_rocm_home() -> Optional[str]:
     return rocm_home
 
 def _find_sycl_home() -> Optional[str]:
```

Comment:
Meanwhile, could we try `ctypes.util.find_library` to find `libsycl.so` and cache the result? `find_library` will simplify the branch code.

Reply:
`ctypes.util.find_library` doesn't return the full path; it returns just the library name (e.g., libze_loader.so). It's ok to check level_zero installation but not suitable for locating sycl_home.

Reply:
If `find_library` does not return `None`, the given library has been placed in the system path. It means that we can link the given library directly without giving the library path. Right? How to find sycl files will be the only open that needs to be resolved.

Reply:
Yes, find_library use dl_open implementation.

Reply:
By the way, we can simplify the logic with some valid assumptions

- For the development, we assume users get Intel GPU dev SDK installed and the `SYCL_HOME` should have been set properly. Otherwise, the torch application cannot even run.
- For the deployment, we assume users get Intel GPU dev SDK python package installed. So, we just need to take the python packages into account.

Does it make sense? @etaf